MeshGeoToolsLib/ConstructMeshesFromGeometries.cpp

```diff
@@ -0,0 +1,114 @@
+/**
+ * \file
+ * \copyright
+ * Copyright (c) 2012-2018, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ */
+
+#include "ConstructMeshesFromGeometries.h"
+
+#include <logog/include/logog.hpp>
+
+#include "MeshLib/Elements/Element.h"
+#include "MeshLib/Mesh.h"
+#include "MeshLib/MeshEditing/DuplicateMeshComponents.h"
+#include "MeshLib/Node.h"
+
+#include "BoundaryElementsSearcher.h"
+#include "MeshNodeSearcher.h"
+
+namespace MeshGeoToolsLib
+{
+template <typename GeometryVec>
+std::vector<MeshLib::Mesh*> constructAdditionalMeshesFromGeometries(
+    std::vector<GeometryVec*> const& geometries,
+    MeshGeoToolsLib::BoundaryElementsSearcher& boundary_element_searcher)
+{
+    std::vector<MeshLib::Mesh*> additional_meshes;
+
+    for (GeometryVec* const geometry_vec : geometries)
+    {
+        // Each geometry_vec has a name, this is the first part of the full
+        // name.
+        auto const& vec_name = geometry_vec->getName();
+
+        auto const& vec_data = *geometry_vec->getVector();
+
+        auto const vec_size = geometry_vec->size();
+        for (std::size_t i = 0; i < vec_size; ++i)
+        {
+            // Each geometry has a name, this is the second part of the full
+            // name.
+            std::string geometry_name;
+            bool const is_geometry_named =
+                geometry_vec->getNameOfElementByID(i, geometry_name);
+            if (!is_geometry_named)
+            {
+                continue;
+            }
+
+            auto const& geometry = *vec_data[i];
+
+            DBUG("Creating mesh from geometry %s %s.", vec_name.c_str(),
+                 geometry_name.c_str());
+
+            auto mesh = createMeshFromElementSelection(
+                vec_name + "_" + geometry_name,
+                MeshLib::cloneElements(
+                    boundary_element_searcher.getBoundaryElements(geometry)));
+
+            additional_meshes.push_back(mesh.release());
+        }
+    }
+    return additional_meshes;
+}
+
+std::vector<MeshLib::Mesh*> constructAdditionalMeshesFromGeoObjects(
+    GeoLib::GEOObjects const& geo_objects, MeshLib::Mesh const& mesh)
+{
+    std::vector<MeshLib::Mesh*> additional_meshes;
+
+    // TODO (naumov) add config for search length algorithms.
```

Comment:
I'm curious how long it takes until the first users will complain, because their BCs are too "shitty" :smile: